flink-yarn/src/main/java/org/apache/flink/yarn/WorkerSpecContainerResourceAdapter.java

```diff
@@ -122,8 +128,8 @@ private InternalContainerResource createAndMapContainerResource(final WorkerReso
 		final TaskExecutorProcessSpec taskExecutorProcessSpec =
 			TaskExecutorProcessUtils.processSpecFromWorkerResourceSpec(flinkConfig, workerResourceSpec);
 		final InternalContainerResource internalContainerResource = new InternalContainerResource(
-			normalize(taskExecutorProcessSpec.getTotalProcessMemorySize().getMebiBytes(), minMemMB),
-			normalize(taskExecutorProcessSpec.getCpuCores().getValue().intValue(), minVcore),
+			normalize(taskExecutorProcessSpec.getTotalProcessMemorySize().getMebiBytes(), unitMemMB),
+			normalize(taskExecutorProcessSpec.getCpuCores().getValue().intValue(), unitVcore),
```

Comment:
This should not work.
The method `normalize` should take both unit and min values into consideration. With the current implementation, there's a change that the normalized resource is smaller than min.

Reply:
I think this also reveals that we have failed cover such scenarios with our test cases.